pytradfri/smart_task.py

```diff
@@ -227,47 +284,53 @@ def __init__(self, task, index, state: bool, path, raw):
         self._raw = raw
 
     @property
-    def devices_dict(self):
-        """Return state of start action task."""
-        json_list = {}
-        index = 0
-        for x_entry in self._raw[ROOT_START_ACTION]:
-            if index != self.index:
-                json_list.update(x_entry)
-            index = index + 1
-        return json_list
+    def devices_list(self) -> list[dict[str, int]]:
+        """Store task data for all tasks but the one we want to update."""
+        output_list: list[dict[str, int]] = []
+        current_data_list: list[StartActionResponse] = self._raw.root_start_action
+        for idx, record in enumerate(current_data_list):
+            if idx != self.index:
+                output_list.append(
+                    {
+                        ATTR_ID: record.id,
+                        ATTR_TRANSITION_TIME: record.transition_time,
+                        ATTR_LIGHT_DIMMER: record.dimmer,
+                    }
+                )
+
+        return output_list
 
     @property
-    def id(self):
+    def id(self) -> int:
         """Return ID (device id) of task."""
-        return self.raw.get(ATTR_ID)
+        return self.raw.id
 
     @property
-    def item_controller(self):
+    def item_controller(self) -> StartActionItemController:
         """Control a task."""
         return StartActionItemController(
-            self, self.raw, self.state, self.path, self.devices_dict
+            self, self.raw, self.state, self.path, self.devices_list
         )
 
     @property
-    def transition_time(self):
+    def transition_time(self) -> int:
         """Transition runs for this long from the time in task_start.
 
         Value is in seconds x 10. Default to 0 if transition is missing.
         """
-        return self.raw.get(ATTR_TRANSITION_TIME, 0) / 60 / 10
+        return round(self.raw.transition_time / 60 / 10)
```

Comment:
Are we sure that transition_time is not optional?

See https://github.com/home-assistant-libs/pytradfri/pull/276

Reply:
Oh, I looked at the sample data in the test folder. But you have digged into this more recently so I believe you are right.